client/my-sites/plans/jetpack-plans/product-grid/utils.ts

```diff
@@ -90,7 +90,11 @@ export const getProductsToDisplay = ( {
 		.filter( ( product ): product is SelectorProduct => product?.term === duration )
 		// Remove duplicates (only happens if the site somehow has the same product
 		// both purchased and included in a plan, very unlikely)
-		.filter( ( product ) => ! purchasedSlugs.includes( product.productSlug ) );
+		.filter(
+			( product ) =>
+				! purchasedSlugs.includes( product.productSlug ) &&
+				! purchasedSlugs.includes( product.monthlyProductSlug )
```

Comment:
What happens if `monthlyProductSlug` is undefined, for instance it's already a monthly product? Would there ever be a case of false matching?

Reply:
Good catch -- I'll add some checks in this predicate before merging. 👍